test/matrix/matrix.cpp

```diff
@@ -0,0 +1,782 @@
+    {
+        ASSERT_TRUE(dynamic_cast<const matrix_type::merge_path*>(
+            mtx->get_strategy().get()));
+    }
+};
+
+struct CsrWithSparselibStrategy
+    : SimpleMatrixTest<gko::matrix::Csr<double, int>> {
+    static std::unique_ptr<matrix_type> create(
+        std::shared_ptr<gko::Executor> exec, gko::dim<2> size)
+    {
+        return matrix_type::create(exec, size, 0,
+                                   std::make_shared<matrix_type::sparselib>());
+    }
+
+    static void check_property(const std::unique_ptr<matrix_type>& mtx)
+    {
+        ASSERT_TRUE(dynamic_cast<const matrix_type::sparselib*>(
+            mtx->get_strategy().get()));
+    }
+};
+
+struct CsrWithLoadBalanceStrategy
+    : SimpleMatrixTest<gko::matrix::Csr<double, int>> {
+    static std::unique_ptr<matrix_type> create(
+        std::shared_ptr<gko::Executor> exec, gko::dim<2> size)
+    {
+        return matrix_type::create(exec, size, 0,
+                                   std::make_shared<matrix_type::load_balance>(
+                                       gko::EXEC_TYPE::create(0, exec)));
+    }
+
+    static void check_property(const std::unique_ptr<matrix_type>& mtx)
+    {
+        ASSERT_TRUE(dynamic_cast<const matrix_type::load_balance*>(
+            mtx->get_strategy().get()));
+    }
+};
+
+struct CsrWithAutomaticalStrategy
+    : SimpleMatrixTest<gko::matrix::Csr<double, int>> {
+    static std::unique_ptr<matrix_type> create(
+        std::shared_ptr<gko::Executor> exec, gko::dim<2> size)
+    {
+        return matrix_type::create(exec, size, 0,
+                                   std::make_shared<matrix_type::automatical>(
+                                       gko::EXEC_TYPE::create(0, exec)));
+    }
+
+    static void check_property(const std::unique_ptr<matrix_type>& mtx)
+    {
+        ASSERT_TRUE(dynamic_cast<const matrix_type::automatical*>(
+            mtx->get_strategy().get()));
+    }
+};
+
+
+#endif
+
+
+struct Ell : SimpleMatrixTest<gko::matrix::Ell<double, int>> {};
+
+struct FbcsrBlocksize1 : SimpleMatrixTest<gko::matrix::Fbcsr<double, int>> {
+    static bool preserves_zeros() { return false; }
+
+    static std::unique_ptr<matrix_type> create(
+        std::shared_ptr<gko::Executor> exec, gko::dim<2> size)
+    {
+        return matrix_type::create(exec, size, 0, 1);
+    }
+
+    static void check_property(const std::unique_ptr<matrix_type>& mtx)
+    {
+        ASSERT_EQ(mtx->get_block_size(), 1);
+    }
+};
+
+struct FbcsrBlocksize2 : SimpleMatrixTest<gko::matrix::Fbcsr<double, int>> {
+    static bool preserves_zeros() { return false; }
+
+    static std::unique_ptr<matrix_type> create(
+        std::shared_ptr<gko::Executor> exec, gko::dim<2> size)
+    {
+        return matrix_type::create(exec, size, 0, 2);
+    }
+
+    static void check_property(const std::unique_ptr<matrix_type>& mtx)
+    {
+        ASSERT_EQ(mtx->get_block_size(), 2);
+    }
+};
+
```

Comment:
```suggestion


```
I guess you split different format with 2 empty line but different setting with 1 empty line?

Reply:
yes